frontend/src/Dashboard/CrossingListPage/CrossingListItem/CrossingListItem.js

```diff
@@ -119,67 +127,103 @@ class CrossingListItem extends React.Component {
-              <input className="NotesTextBox" type="text" value={this.state.notes} onChange={this.notesChanged}/>
-            </div>
+          <Dropdown
+            options={reasons.filter(reason => reason.statusId === this.state.selectedStatus)}
+            selected={this.state.selectedReason}
+            onChange={this.reasonChanged} />
+        </div>
+      ) : (
+        <div className="CrossingListItemFlexItem--spacer" />
+      )}
+        <div className="CrossingListItemFlexItem">
+          <div className="ControlLabel">Notes to the public</div>
+          <input className="NotesTextBox" type="text" value={this.state.notes} onChange={this.notesChanged}/>
+        </div>
+      </div>
+
+      
+      { (show.includes('duration') || show.includes('cancelSave')) && (
+
+      <div className="CrossingListItemFlexContainer">
+        <div className="CrossingListItemFlexItem--spacer" />
+        
+      {show.includes('duration') ? (
+        <div className="CrossingListItemFlexItem">
+          <div className="ControlLabelContainer">
+            <div className="ControlLabel">Duration</div>
+            <div className="required">{this.isDirty() ? "Required" : ""}</div>
           </div>
-          <div className={show.includes('duration') || show.includes('cancelSave') ? "CrossingListItemFlexContainer" : ""}>
-            <div className="flexitem" />
-            <div className="flexitem">
-              <div className={show.includes('duration') ? "" : "hidden"}>
-                <div className="ControlLabelContainer">
-                  <div className="ControlLabel">Duration</div>
-                  <div className="required">{this.isDirty() ? "Required" : ""}</div>
-                </div>
-                <Dropdown
-                  options={durations}
-                  selected={this.state.selectedDuration}
-                  onChange={this.durationChanged} />
-              </div>
-            </div>
-            <div className="flexitem">
-              <div className={show.includes('cancelSave') ? "" : "hidden"}>
-                <div className="flexcontainer">              
-                  <div className="CancelButton" onClick={this.cancelClicked}>Cancel</div>
-                  <div className="SaveButton" onClick={this.newStatusUpdate.bind(this)}>Save</div>
-                </div>
-              </div>
-            </div>
+          <Dropdown
+            options={durations}
+            selected={this.state.selectedDuration}
+            onChange={this.durationChanged} />
+        </div>
+      ) : (
+        <div className="CrossingListItemFlexItem--spacer" />
+      )}
+
+      {show.includes('cancelSave') ? (
+        <div className="CrossingListItemFlexItem">
+          <div className="ButtonContainer">              
+            <div className="CancelButton" onClick={this.cancelClicked}>Cancel</div>
+            <div className="SaveButton" onClick={this.newStatusUpdate.bind(this)}>Save</div>
```

Comment:
These button classnames seems like pretty universal. Should we refactor towards BEM? Ex: `CrossingListItem--saveButton`. Should we also extract out the button mixin so we can use these button styles in more places?